dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraph.java

```diff
@@ -99,6 +99,13 @@ void addPath(DependencyPath path) {
   public List<DependencyPath> list() {
     return new ArrayList<>(graph);
   }
+  
+  /**
+   * @return a mutable copy of the artifacts in this graph in breadth first order
```

Comment:
```suggestion
   * Returns a the artifacts in this graph in breadth first order
```

Is mutability important? If not, let's use ImmutableList.

Reply:
Clarify that this list may contain duplicate; the caller is responsible to perform dependency mediation.

Reply:
The current implementation returns a list that explicitly has no guarantees on mutability.

Reply:
It currently may return duplicates, but eventually we wouldn't want it to, right?

Reply:
I suppose that depends on where the dependency mediation is being performed. If we can move that completely into this class and have this class return only a classpath, then yes. If dependency mediation happens outside of this class, then no.